pycheribuild/projects/cross/rtems.py

```diff
@@ -0,0 +1,69 @@
+#-
+# SPDX-License-Identifier: BSD-2-Clause
+#
+# Author: Hesham Almatary <Hesham.Almatary@cl.cam.ac.uk>
+#
+# This software was developed by SRI International and the University of
+# Cambridge Computer Laboratory (Department of Computer Science and
+# Technology) under DARPA contract HR0011-18-C-0016 ("ECATS"), as part of the
+# DARPA SSITH research programme.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE AUTHOR AND CONTRIBUTORS ``AS IS'' AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+# OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+#
+from .crosscompileproject import *
+from ..project import *
+from ...utils import IS_MAC, runCmd
+
+class BuildRtems(CrossCompileProject):
+    repository = GitRepository("https://github.com/CTSRD-CHERI/rtems",
+        per_target_branches={
+            CompilationTargets.RTEMS_RISCV64_PURECAP: TargetBranchInfo("cheri_waf1", "rtems-riscv")
+            })
+    target = "rtems"
```

Comment:
Are there any dependencies? newlib or compiler-rt? Or are they bundled with RTEMS?
Something like this?
```suggestion
    target = "rtems"
    dependencies ["newlib-rtems-riscv64-purecap", "compiler-rt-builtins-rtems-riscv64-purecap"]
```

Reply:
That's bundled part of _RTEMS_RISCV64_PURECAP_ and _RTEMSTargetInfo_.

Reply:
Ah I see now why you have the separate target_infos. It should be possible to add `is_sdk_target = True` to avoid those, but I'll take care of that cleanup later.